room/staff.py

```diff
@@ -136,6 +136,28 @@ def get(self, id=None):
     return common.Respond(self.request, 'site_list_one', d)
 
 
+class SiteLookup(StaffHandler):
+  def get(self, site_number=None):
+    if site_number is not None:
+      query = ndb_models.NewSite.query(ndb_models.NewSite.number == site_number)
+      results = list(query)
+      if not results:
+        return self.redirect_to('NoSiteFound', bad_site_number=site_number)
```

Comment:
You can just write the not found output here.   Can simplify and scrap the extra handler.

Reply:
good call